From ab4b8bd8e262e226fe40d022f7a46d9a9758bc1f Mon Sep 17 00:00:00 2001
From: Petr Viktorin <pviktori@redhat.com>
Date: Fri, 6 Jun 2014 14:21:16 +0200
Subject: [PATCH 109/116] ipalib.frontend: Do API version check before
 converting arguments

This results in the proper message being shown if the client sends
an option the server doesn't have yet.

https://fedorahosted.org/freeipa/ticket/3963

Reviewed-By: Martin Kosek <mkosek@redhat.com>
---
 ipalib/frontend.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ipalib/frontend.py b/ipalib/frontend.py
index fadcb8632cab9d7a49aacc72d9e804ba7766f33a..57713c192baf22c24d8f95e01a5d52f41634a1ec 100644
--- a/ipalib/frontend.py
+++ b/ipalib/frontend.py
@@ -418,6 +418,9 @@ def __call__(self, *args, **options):
         XML-RPC and the executed an the nearest IPA server.
         """
         self.ensure_finalized()
+        version_provided = 'version' in options
+        if version_provided:
+            self.verify_client_version(unicode(options['version']))
         params = self.args_options_2_params(*args, **options)
         self.debug(
             'raw: %s(%s)', self.name, ', '.join(self._repr_iter(**params))
-- 
1.9.3

